examples/benchmark/datagen.py

```diff
@@ -0,0 +1,345 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+import abc
+import io
+import os
+import tarfile
+import time
+from multiprocessing import Queue
+from threading import Thread, Lock, Barrier
+from typing import Any, Callable, Iterator, TypeVar, NewType
+
+import boto3
+import click
+import numpy as np
+import prefixed as pr
+import yaml
+from PIL import Image
+
+
+class DataGenerator(abc.ABC):
+    @abc.abstractmethod
+    def create(self, idx_gen: Iterator[int]) -> Iterator[tuple[str, io.BytesIO]]:
+        pass
+
+
+class ImageGenerator(DataGenerator):
+    def __init__(self, width: int, height: int, img_format: str = "JPEG"):
+        self.width = width
+        self.height = height
+        self.img_format = img_format
+
+    def create(self, idx_iter: Iterator[int]) -> Iterator[tuple[str, io.BytesIO]]:
```

Comment:
We're using `tuple[str, BytesIO]` a few times here. Why not create a named tuple for it, or a dataclass?